src/components/CollaboratorModal/components/MainSubmodal.tsx

```diff
@@ -176,16 +166,23 @@ export const MainSubmodal = ({
   const curCollaboratorValue = collaboratorFormMethods.watch(
     "newCollaboratorEmail"
   )
-
-  useEffect(() => {
-    if (addCollaboratorSuccess) {
+  const {
+    mutate: addCollaborator,
+    isError: isAddCollaboratorError,
+    isLoading: isAddCollaboratorLoading,
+    reset,
+  } = CollaboratorHooks.useAddCollaboratorHook(siteName, {
+    onSuccess: () => {
       successToast({
         id: "add-collaborator",
         description: "Collaborator added successfully",
       })
       collaboratorFormMethods.reset()
-    }
-  }, [addCollaboratorSuccess, collaboratorFormMethods, successToast])
+    },
+    onError: (err) => {
+      setErrorMessage(extractErrorMessage(err.response?.data.error))
+    },
```

Comment:
this honestly feels like a code smell ._. it's not immediately obvious where this error message is used and there doesn't seem to be feedback?